go-controller/pkg/node/gateway_udn.go

```diff
@@ -80,26 +87,9 @@ func (udng *UserDefinedNetworkGateway) DelNetwork() error {
 // STEP2: It saves the MAC address generated on the 1st go as an option on the OVS interface
 // so that it persists on reboots
 // STEP3: sets up the management port link on the host
-// STEP4: adds the management port IP .2 to the mplink
-// STEP5: adds the mac address to the node management port annotation
-func (udng *UserDefinedNetworkGateway) addUDNManagementPort() (netlink.Link, error) {
+func (udng *UserDefinedNetworkGateway) addUDNManagementPort() (netlink.Link, net.HardwareAddr, error) {
```

Comment:
nit: add func comment for what are the things we are returning from this func because that list seems to grow